DQMOffline/Trigger/plugins/JetMETMonitor.cc

```diff
@@ -0,0 +1,1056 @@
+        jetmetHE_EtaME_.numerator -> Fill(jeteta_);
+        jetmetHE_VsLS_.numerator -> Fill(ls, jetpt_);
+        jetmetHE_EtaVsPhi_.numerator -> Fill(jeteta_,jetphi_);
+   
+        jetmetHE_p_ME_.numerator -> Fill(jetpt_);
+        jetmetHE_p_ME_variableBinning_.numerator -> Fill(jetpt_);
+        jetmetHE_p_PhiME_.numerator -> Fill(jetphi_);
+        jetmetHE_p_EtaME_.numerator -> Fill(jeteta_);
+        jetmetHE_p_VsLS_.numerator -> Fill(ls, jetpt_);
+        jetmetHE_p_EtaVsPhi_.numerator -> Fill(jeteta_,jetphi_);
+     }
+     if (isEndCapM( jeteta_ ) )
+     {
+        jetmetHE_ME_.numerator -> Fill(jetpt_);
+        jetmetHE_ME_variableBinning_.numerator -> Fill(jetpt_);
+        jetmetHE_PhiME_.numerator -> Fill(jetphi_);
+        jetmetHE_EtaME_.numerator -> Fill(jeteta_);
+        jetmetHE_VsLS_.numerator -> Fill(ls, jetpt_);
+        jetmetHE_EtaVsPhi_.numerator -> Fill(jeteta_,jetphi_);
+   
+        jetmetHE_m_ME_.numerator -> Fill(jetpt_);
+        jetmetHE_m_ME_variableBinning_.numerator -> Fill(jetpt_);
+        jetmetHE_m_PhiME_.numerator -> Fill(jetphi_);
+        jetmetHE_m_EtaME_.numerator -> Fill(jeteta_);
+        jetmetHE_m_VsLS_.numerator -> Fill(ls, jetpt_);
+        jetmetHE_m_EtaVsPhi_.numerator -> Fill(jeteta_,jetphi_);
+     }
+     if (isForward( jeteta_ ) )
+     {
+        jetmetHF_ME_.numerator -> Fill(jetpt_);
+        jetmetHF_ME_variableBinning_.numerator -> Fill(jetpt_);
+        jetmetHF_PhiME_.numerator -> Fill(jetphi_);
+        jetmetHF_EtaME_.numerator -> Fill(jeteta_);
+        jetmetHF_VsLS_.numerator -> Fill(ls, jetpt_);
+        jetmetHF_EtaVsPhi_.numerator -> Fill(jeteta_,jetphi_);
+     }
+     if (isHEP17( jeteta_, jetphi_ ) )
+     {
+        jetmetHEP17_ME_.numerator -> Fill(jetpt_);
+        jetmetHEP17_ME_variableBinning_.numerator -> Fill(jetpt_);
+        jetmetHEP17_PhiME_.numerator -> Fill(jetphi_);
+        jetmetHEP17_EtaME_.numerator -> Fill(jeteta_);
+        jetmetHEP17_VsLS_.numerator -> Fill(ls, jetpt_);
+        jetmetHEP17_EtaVsPhi_.numerator -> Fill(jeteta_,jetphi_);
+     }
+     if (isHEM17( jeteta_, jetphi_ ) )
+     {
+        jetmetHEM17_ME_.numerator -> Fill(jetpt_);
+        jetmetHEM17_ME_variableBinning_.numerator -> Fill(jetpt_);
+        jetmetHEM17_PhiME_.numerator -> Fill(jetphi_);
+        jetmetHEM17_EtaME_.numerator -> Fill(jeteta_);
+        jetmetHEM17_VsLS_.numerator -> Fill(ls, jetpt_);
+        jetmetHEM17_EtaVsPhi_.numerator -> Fill(jeteta_,jetphi_);
+     }
+     if (isHEP18( jeteta_, jetphi_ ) )
+     {
+        jetmetHEP18_ME_.numerator -> Fill(jetpt_);
+        jetmetHEP18_ME_variableBinning_.numerator -> Fill(jetpt_);
+        jetmetHEP18_PhiME_.numerator -> Fill(jetphi_);
+        jetmetHEP18_EtaME_.numerator -> Fill(jeteta_);
+        jetmetHEP18_VsLS_.numerator -> Fill(ls, jetpt_);
+        jetmetHEP18_EtaVsPhi_.numerator -> Fill(jeteta_,jetphi_);
+     } 
+  }// PFJet Trigger 
+  // CaloJet Trigger
+  if (isCaloJetTrig) {
```

Comment:
couldn't you define a function which handles the histograms
instead of having this copy-n-paste ?

in addition, even if I suggest to develop even further this code allowing for PF specific quantities to be monitored,
as now why don't you make use of the edm::View<reco::Jet> or something like it ?